server/main.py

```diff
@@ -218,38 +202,44 @@ async def room(websocket: WebSocket) -> None:
     """
     client = Client(websocket)
     await client.accept()
+
     try:
-        initial_data = ConnectionData(**await client.receive())
+        initial_event = Event(**await client.receive())
     except WebSocketDisconnect:
         return
 
-    room_code = initial_data.data.room_code
-    if initial_data.type == "connect":
-        match initial_data.data.connection_type:
-            case "create":
-                manager.create_room(client, room_code)
-            case "join":
-                try:
-                    manager.join_room(client, room_code)
-                except RoomNotFoundError as e:
-                    # Send off to frontend to handle
-                    await client.send(
-                        {
-                            "type": "error",
-                            "data": {
-                                "message": e.message,
-                            },
-                            "status_code": StatusCode.ROOM_NOT_FOUND,
-                        }
-                    )
-                    await client.close()
-                    return
-            case _:
-                raise NotImplementedError
+    if initial_event.type != EventType.CONNECT:
+        return
+
+    initial_data: ConnectData = initial_event.data
+    print(initial_event, initial_data)
+    room_code = initial_data.room_code
+
+    match initial_data.connection_type:
+        case "create":
+            manager.create_room(client, room_code)
+        case "join":
+            try:
+                manager.join_room(client, room_code)
+            except RoomNotFoundError as e:
+                await client.send(
+                    Event(type=EventType.ERROR, data=ErrorData(message=e.message), status_code=StatusCode.SUCCESS)
+                )
+                await client.close()
+                return
+
+    manager.broadcast(initial_event, room_code)
 
     try:
         while True:
             data = await client.receive()
```

Comment:
`data = Event(**await client.receive())`

Reply:
Fixed in bdd6848c260861f606d0fe7f3d885646d55fa8e8.